tests/build-tools.test.js

```diff
@@ -0,0 +1,112 @@
+const { readFileSync, rmSync, mkdirSync, writeFileSync } = require('fs');
+const { resolve } = require('path');
+const { getData } = require('../scripts/tools/extract-tools-github');
+const { convertTools } = require('../scripts/tools/tools-object');
+const { combineTools } = require('../scripts/tools/combine-tools');
+const { buildTools } = require('../scripts/build-tools');
+const { tagsData, manualTools, mockConvertedData, initialToolsData, mockExtractData } = require('../tests/fixtures/buildToolsData');
+
+jest.mock('../scripts/tools/extract-tools-github');
+jest.mock('../scripts/tools/tools-object');
+jest.mock('../scripts/tools/combine-tools');
+
+describe('buildTools', () => {
+    const testDir = resolve(__dirname, 'test_config');
+    const toolsPath = resolve(testDir, 'tools.json');
+    const tagsPath = resolve(testDir, 'all-tags.json');
+    const automatedToolsPath = resolve(testDir, 'tools-automated.json');
+    const manualToolsPath = resolve(testDir, 'tools-manual.json');
+
+    beforeAll(() => {
+        mkdirSync(testDir, { recursive: true });
+
+        writeFileSync(manualToolsPath, JSON.stringify(manualTools));
+    });
+
+    afterAll(() => {
+        rmSync(testDir, { recursive: true, force: true });
+    });
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should extract, convert, combine tools, and write to file', async () => {
+        getData.mockImplementation(async () => {
+            writeFileSync(automatedToolsPath, JSON.stringify(mockExtractData));
+            return mockExtractData;
+        });
+
+        convertTools.mockImplementation(async () => {
+            writeFileSync(automatedToolsPath, JSON.stringify(mockConvertedData));
+            return mockConvertedData;
+        });
+
+        combineTools.mockImplementation(async () => {
+            writeFileSync(toolsPath, JSON.stringify(initialToolsData));
+            writeFileSync(tagsPath, JSON.stringify(tagsData));
+            return true;
```

Comment:
[3] here again, you don't need to write anything. And also, do we even need to mock `combineTools`?
It isn't creating any API call, so why to mock it?